src/Package/Impl/Options/R/Tools/EncodingTypeConverter.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using Microsoft.Common.Core;
+
+namespace Microsoft.VisualStudio.R.Package.Options.R.Tools {
+    internal sealed class EncodingTypeConverter : TypeConverter {
+        public override bool GetStandardValuesSupported(ITypeDescriptorContext context) {
+            return true;
+        }
+
+        public override StandardValuesCollection GetStandardValues(ITypeDescriptorContext context) {
+            var codePageList = new List<int>() { 0 };
+            var codePages = Encoding.GetEncodings()
+                                    .OrderBy(e => e.DisplayName)
+                                    .Select(e => e.CodePage)
+                                    .Where(cp => cp != Encoding.UTF7.CodePage &&
```

Comment:
This would be more readable as a static readonly array, and you can then do `.Except` instead of `.Where`